internal/controller/tokenprovider/token_provider.go

```diff
@@ -0,0 +1,40 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package tokenprovider
+
+import (
+	"context"
+	"time"
+)
+
+type TokenExpiry struct {
+	Token     string
+	ExpiresAt time.Time
+}
+
+type TokenProvider interface {
+	GetToken(ctx context.Context) (TokenExpiry, error)
+}
```

Comment:
could you make sure **all** public structs have comments on both structs as well as methods just like elsewhere